x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1137,6 +1142,51 @@ protected Expression rule(Expression e) {
         }
     }
 
+    public static final class BubbleUpNegations extends OptimizerExpressionRule {
```

Comment:
Kept the `BubbleUpNegations` naming, but not strongly attached to it -- will rename if anyone feels significantly better about `SimplifyNegations`.